arch/arc/Kconfig

```diff
@@ -147,6 +177,7 @@ config ARC_FIRQ_STACK_SIZE
 
 config ARC_HAS_STACK_CHECKING
 	bool "ARC has STACK_CHECKING"
+	depends on ISA_ARCV2
```

Comment:
No stack checking planned at all for V3, or no support in current implementation(s)?
For arch level kconfigs, I think the 'depends' should reflect the ARCv3 ISA, not implementations.

Reply:
I didn't check all, but maybe there are others more relevant to add a 'depend', e.g. HAS_SECURE.

Reply:
> I didn't check all, but maybe there are others more relevant to add a 'depend', e.g. HAS_SECURE.

Yep, HAS_SECURE dependency was missing, now it's fixed, rhanks!

Reply:
> No stack checking planned at all for V3, or no support in current implementation(s)?

No support in current implementation

> For arch level kconfigs, I think the 'depends' should reflect the ARCv3 ISA, not implementations.

I tend to disagree here. We describe the Zephyr configuration via Kconfig (and not a ISA or exact CPU documentation). We shouldn't allow to user to enable feature combination we don't support - it will be confusing and not user-friendly.

Reply:
I guess what triggered my remark was more that if the hardware supports it, and Zephyr supports it for other ARC variants, we should support it for ARCV3 as well. So it depends on timing: in this PR it is not supported, in a next one it probably will.